redbot/cogs/warnings/warnings.py

```diff
@@ -342,36 +341,52 @@ def __init__(self, bot: Red):
 
     @commands.command()
     @commands.guild_only()
-    async def warnings(
-        self, ctx: commands.Context, user: Optional[Union[discord.Member, int]] = None
-    ):
-        """List the warnings for the specified user.
-
-        Omit `<user>` to see your own warnings.
+    @checks.admin()
+    async def warnings(self, ctx: commands.Context, user: Union[discord.Member, int]):
+        """List the warnings for the specified user."""
 
-        Note that showing warnings for users other than yourself requires
-        appropriate permissions.
-        """
-        if user is None:
-            user = ctx.author
-        else:
-            if not await is_admin_or_superior(self.bot, ctx.author):
-                return await ctx.send(
-                    warning(_("You are not allowed to check warnings for other users!"))
+        msg = ""
+        member_settings = self.config.member(user)
+        async with member_settings.warnings() as user_warnings:
+            if not user_warnings.keys():  # no warnings for the user
+                await ctx.send(_("That user has no warnings!"))
+            else:
+                for key in user_warnings.keys():
+                    mod_id = user_warnings[key]["mod"]
+                    mod = ctx.bot.get_user(mod_id) or _("Unknown Moderator ({})").format(mod_id)
+                    msg += _(
+                        "{num_points} point warning {reason_name} issued by {user} for "
+                        "{description}\n"
+                    ).format(
+                        num_points=user_warnings[key]["points"],
+                        reason_name=key,
+                        user=mod,
+                        description=user_warnings[key]["description"],
+                    )
+                await ctx.send_interactive(
+                    pagify(msg, shorten_by=58), box_lang=_("Warnings for {user}").format(user=user)
                 )
 
+    @commands.command()
+    @commands.guild_only()
+    async def mywarnings(self, ctx: commands.Context):
+        """List warnings for yourself."""
+
+        user = ctx.author
+
+        msg = ""
+        member_settings = self.config.member(user)
+        async with member_settings.warnings() as user_warnings:
+
             try:
```

Comment:
Not exactly what I meant... 

The git diff here is doing some slight disfavors in discussing review as it's recognizing the code for `mywarnings` now matches the old `warnings` more closely.

The handling for if a user id not a user was passed isn't needed in `mywarnings` but is needed in `warnings` (prior to interacting with config)